onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -530,6 +592,31 @@ class ReduceAggregatorMin : public ReduceAggregator<T, TVAL> {
           }
         });
   }
+
+  static void FastReduceRKR(const Tensor& input, const std::vector<int64_t>& fast_shape,
+                            Tensor& output, concurrency::ThreadPool* tp) {
+    const T* data = input.Data<T>();
+    T* out = output.MutableData<T>();
+    int64_t d0 = fast_shape[0];
+    int64_t d2 = fast_shape[2];
+    int64_t inc = d2 * fast_shape[1];
+
+    concurrency::ThreadPool::TryParallelFor(
+        tp, fast_shape[1], ParallelReduceFastCost(fast_shape[1], fast_shape[0] * fast_shape[2], sizeof(T), 6),
+        [data, out, d0, d2, inc](ptrdiff_t begin, ptrdiff_t last) {
+          const T* p;
+          for (ptrdiff_t d = begin; d < last; ++d) {
+            p = data + d * d2;
+            out[d] = p[0];
+            for (int64_t i = 0; i < d0; ++i) {
+              for (int64_t k = 0; k < d2; ++k) {
+                out[d] = p[k] < out[d] ? p[k] : out[d];
+              }
```

Comment:
Can we reduce the binary size cost by abstracting out a shared implementation that takes a functor for this check/assignment (i.e. plug in whether it's min/max/sum via the functor), or do we lose the perf gain if we do that?

Reply:
It seems doable. I need to test. I may have to keep one loop in the functor.

Reply:
done